src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml

```diff
@@ -264,6 +264,25 @@ let setup_daemon logger =
             in the gossip network (default: %d)"
            Cli_lib.Default.min_connections)
       (optional int)
+  (* TODO uncomment after introducing Bitswap-based block retrieval *)
+  (* and pubsub_v1 =
+     flag "--pubsub-v1" ~aliases:[ "pubsub-v1" ]
+       ~doc:
+         (Printf.sprintf
+            "Mode of handling pubsub v1 topic: 'ro', 'rw' or 'none' (default: \
+             %s)"
+            (Cli_lib.Arg_type.pubsub_topic_mode_to_string
+               Cli_lib.Default.pubsub_v1))
+       (optional pubsub_topic_mode) *)
+  and pubsub_v0 =
+    flag "--pubsub-v0" ~aliases:[ "pubsub-v0" ]
+      ~doc:
+        (Printf.sprintf
+           "Mode of handling pubsub v0 topic: 'ro', 'rw' or 'none' (default: \
```

Comment:
I'm not sure that I'm comfortable with this, since choosing anything other than `rw` is essentially a unilateral hard-fork on the part of the node.
When @nholland94 and I discussed previously, my understanding was that we wanted a version negotiation like:
* If a peer `p` supports the latest `V_n` gossip topic, remove them from all older ones
* If `p` supports `V_{n-1}`, remove them from all older ones and `V_n`
* ...
* If `p` supports `V_0`, remove them from all newer ones.

Then, every peer is still gossiping to every other peer, but as we upgrade the system they can opt into more efficient protocols by upgrading, but still talk to old/other clients that don't have the new protocol

Reply:
P.S. I see it as near all nodes having `rw` set for both old and new topics. And then libp2p layer will handle topic negotiation and make nodes use the newest topic set available for both.